taichi_3d_gaussian_splatting/GaussianPointCloudRasterisation.py

```diff
@@ -619,10 +626,14 @@ def gaussian_point_rasterisation_backward(
                     alpha_grad: ti.f32 = alpha_grad_from_rgb.sum()
                     if enable_depth_grad:
                         depth_i = tile_point_depth[idx_point_offset_with_sort_key_in_block]
-                        alpha_grad_from_depth = (depth_i * T_i - depth_w_i / (1. - alpha)) \
-                            * pixel_depth_grad
+                        alpha_grad_from_depth = (T_i * (depth_i-d_pixel) + 1.0/(1.0-alpha) * \
+                                                 (depth_w_i -  acc_alpha_w_i* d_pixel))  / \
+                        (org_accumulated_alpha+0.00001) * pixel_depth_grad
+                        alpha_grad_from_accumulated_alpha = (T_i- 1.0/(1.0-alpha) * acc_alpha_w_i) *                        _accumulated_alpha_grad
```

Comment:
Please format your code. I haven't looked deep into your formula so I'm not sure if it is correct, but for implementation, it seems to be the same as your formula. Have you gotten it to converge on alpha in some simple tests?

Reply:
Formated the code, hopefully it's now easier to understand.

> Have you gotten it to converge on alpha in some more complex tests? e.g. as an extra loss?

What do you mean as an "extra loss"?

Reply:
Yep